File: csharp/main/TimeDef.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class TimeDef : IConstantScope, ITypeDef<TypeTime>
	{
		private TimeDef()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeTime.Instance, new LiteralTime(LiteralTime.MinValueAsString)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeTime.Instance, new LiteralTime(LiteralTime.MaxValueAsString)));
		}

		public static TimeDef Instance = new TimeDef();

		public String Name => "Time";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeTime Type => TypeTime.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}

File: java/src/main/java/deltix/luminary/TimeDef.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class TimeDef implements TypeDef<TypeTime>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private TimeDef() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeTime.INSTANCE, new LiteralTime(LiteralTime.MIN_VALUE_AS_STRING)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeTime.INSTANCE, new LiteralTime(LiteralTime.MAX_VALUE_AS_STRING)));
    }

    public static final TimeDef INSTANCE = new TimeDef();

    @Override
    public String getName() {
        return "Time";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeTime getType() {
        return TypeTime.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}
